From 8a4d78e71f24628b43bc58675e8348eabd8537c3 Mon Sep 17 00:00:00 2001
From: Arnaldo Garcia Rincon <agarrcia@amazon.com>
Date: Thu, 26 Mar 2026 18:29:59 +0000
Subject: [PATCH] Revert "kbuild: hide tools/ build targets from external
 module builds"

This reverts commit 1bb0b18a06dceee1fdc32161a72e28eab6f011c4 in which
the targets to build "tools/*" were hidden for external modules, but
they are required by the kmod kit since the 'tools/*' binaries are not
distributed as part of the archive.

Signed-off-by: Arnaldo Garcia Rincon <agarrcia@amazon.com>
[jepiyush: adjust context for 6.12.77]
Signed-off-by: Piyush Jena <jepiyush@amazon.com>
---
 Makefile | 16 +++++++++-------
 1 file changed, 9 insertions(+), 7 deletions(-)

diff --git a/Makefile b/Makefile
index e4aa2e76ea56..9089203c7e20 100644
--- a/Makefile
+++ b/Makefile
@@ -1449,13 +1449,6 @@ ifneq ($(wildcard $(objtool_O)),)
 	$(Q)$(MAKE) -sC $(abs_srctree)/tools/objtool O=$(objtool_O) srctree=$(abs_srctree) clean
 endif

-tools/: FORCE
-	$(Q)mkdir -p $(objtree)/tools
-	$(Q)$(MAKE) O=$(abspath $(objtree)) subdir=tools -C $(srctree)/tools/
-
-tools/%: FORCE
-	$(Q)mkdir -p $(objtree)/tools
-	$(Q)$(MAKE) O=$(abspath $(objtree)) subdir=tools -C $(srctree)/tools/ $*

 # ---------------------------------------------------------------------------
 # Kernel selftest
@@ -2144,6 +2137,15 @@ kernelversion:
 image_name:
 	@echo $(KBUILD_IMAGE)

+tools/: FORCE
+	$(Q)mkdir -p $(objtree)/tools
+	$(Q)$(MAKE) O=$(abspath $(objtree)) subdir=tools -C $(srctree)/tools/
+
+tools/%: FORCE
+	$(Q)mkdir -p $(objtree)/tools
+	$(Q)$(MAKE) O=$(abspath $(objtree)) subdir=tools -C $(srctree)/tools/ $*
+
+
 PHONY += run-command
 run-command:
 	$(Q)$(KBUILD_RUN_COMMAND)
--
2.52.0

